src/taskgraph/optimize/base.py

```diff
@@ -307,7 +335,13 @@ def replace_tasks(
             deadline = max(
                 resolve_timestamps(now, task.task["deadline"]) for task in dependents
             )
-        repl = opt.should_replace_task(task, params, deadline, arg)
+        if isinstance(opt, IndexSearch):
+            repl = opt.should_replace_task(
+                task, params, deadline, arg, index_to_taskid, taskid_to_status
+            )
+        else:
+            repl = opt.should_replace_task(task, params, deadline, arg)
+
```

Comment:
I'd prefer to leave the interface unchanged here, so I suggest you do something like:
```
if isinstance(opt, IndexSearch):
    arg = (arg, index_to_task_id, taskid_to_status)

repl = opt.should_replace_task(task, params, deadline, arg)
```

Then unpack `arg` in the `IndexSearch` strategy correspondingly. We use a similar technique in a few other places so this would be a bit more consistent.